lib/novu/http.ex

```diff
@@ -0,0 +1,44 @@
+defmodule Novu.Http do
+  @moduledoc """
+  Helper functions for interacting with Req.
+  """
+  require Logger
+
+  @api_key Application.compile_env(:novu, :api_key)
+  @base_domain Application.compile_env(:novu, :domain, "")
+  @user_agent "Novu/" <> Mix.Project.config()[:version]
+
+  def delete(url) do
+    url
+    |> build_req()
+    |> Req.delete!()
+    |> handle_response()
+  rescue
+    e -> handle_response(e)
+  end
+
+  def post(url, body) do
+    url
+    |> build_req()
+    |> Req.post!(json: body)
+    |> handle_response()
+  rescue
+    e -> handle_response(e)
+  end
+
+  defp build_req(url),
+    do: Req.new(base_domain: @base_domain, headers: request_headers(), url: url, user_agent: @user_agent)
+
+  defp handle_response(%{body: body, status: status_code}) when status_code in 200..299, do: {:ok, body}
+  defp handle_response(%{body: %{"errors" => errors}}), do: {:error, errors}
+  defp handle_response(%Mint.TransportError{reason: :timeout}), do: {:error, :timeout}
+  defp handle_response(response), do: {:error, response}
+
+  defp request_headers do
+    :otel_propagator_text_map.inject([
```

Comment:
We shouldn't be sending open telemetry headers to Novu as it's not the same infrastructure. I think best we can do here is add a bunch of telemetry points. If we _wanted_ we could add an open telemetry module that listens to those telemetry points and makes spans off them.

Reply:
Yup! Removed. This was left over from copying things from an existing client.